src/broadcaster/accounts.rs

```diff
@@ -0,0 +1,46 @@
+use crate::broadcaster::clients::AccountQueryClient;
+use cosmos_sdk_proto::cosmos::auth::v1beta1::{BaseAccount, QueryAccountRequest, QueryAccountResponse};
+use cosmos_sdk_proto::traits::Message;
+use error_stack::{FutureExt, IntoReport, Result, ResultExt};
+use thiserror::Error;
+
+use crate::url::Url;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("failed to connect to node {url}")]
+    ConnectionFailed { url: Url },
+    #[error("failed to retrieve the account information for address {address}")]
+    AccountResponseFailed { address: String },
+    #[error("address {address} is unknown")]
+    AccountNotFound { address: String },
+    #[error("failed to unmarshal account response")]
+    UnmarshalingFailed,
+}
+
+pub async fn account<T>(mut client: T, address: String) -> Result<BaseAccount, Error>
+where
+    T: AccountQueryClient,
+{
+    let response = client
+        .account(QueryAccountRequest {
+            address: address.clone(),
+        })
+        .change_context(Error::AccountResponseFailed {
+            address: address.clone(),
+        })
```

Comment:
```suggestion
        .change_context_lazy(|| Error::AccountResponseFailed {
            address: address.clone(),
        })
```
For error that contains extra information, I'd personally prefer doing it lazily.